src/main/java/com/epam/aidial/core/controller/ResourceController.java

```diff
@@ -158,9 +164,12 @@ private Future<?> deleteResource(ResourceDescription descriptor) {
         return vertx.executeBlocking(() -> {
                     Set<ResourceLink> resourceLinks = new HashSet<>();
                     resourceLinks.add(new ResourceLink(descriptor.getUrl()));
-                    shareService.revokeSharedAccess(descriptor.getBucketName(), descriptor.getBucketLocation(),
-                            new ResourceLinkCollection(resourceLinks));
-                    return service.deleteResource(descriptor);
+                    return lockService.underUserLock(proxy, context, () -> {
```

Comment:
We're going to use multiple level locks:

1. User level lock everything will be locked under /Users/<user>
2. Resource level lock

Could you explain why we need both lock levels?
User level lock is a strong lock which could prevent a concurrent access to different resources of the same user. We need to understand reasons of doing that.